### PR TITLE
Error checking

### PR DESCRIPTION
**Pull request by Edih112**
## What
Adds basic support for static semantic checking and error reporting through error underlines and hover information in the editor.
## How
The main parts are the SymbolTableVisitor, which builds up a symbol table of the program, and then the SemanticVisitor, which uses the symbol table to do more specific error checking. The other files all help to support these two steps. Finally, the ModelSemanticChecker utilizes both in order to perform one round of error checking on the current program state.
## Testing
The changes have accompanying tests for errors detected by the symbol table visitor, as well as tests for checking for relationships between types.

### CODE REVIEW

File: src/language-handler/SemanticVisitor.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+   * @param ctx context of an variable
+   * @requires ctx does not have an exception.
+   */
+  private varInitializationCheck(ctx: SpeciesContext | Var_nameContext) {
+    const currST: SymbolTable | undefined = this.getCurrST();
+    const varName = ctx.NAME().text;
+    const idSrcRange: SrcRange = this.getSrcRange(ctx.NAME());
+    if (currST) {
+      const varInfo = currST.getVar(varName);
+      if (this.currNameAndScope?.scope === 'function') {
```
Comment (evaxliu) [inline src/language-handler/SemanticVisitor.ts:98]: Empty if statement, will this contain anything later?

**Review by evaxliu — COMMENTED** [review]
Overall looks very good. Had a small comment for you to review over, if all is fine I can approve.

File: src/language-handler/SemanticVisitor.ts (hunk 1 above)
Comment (Edih112) [reply src/language-handler/SemanticVisitor.ts:98]: ah that doesn't need to be there, I'll change that real quick.

**Review by evaxliu — APPROVED** [review]
LGTM! Good work.

**Pull request merged**